pele_platform/Utilities/Helpers/map_atoms.py

```diff
@@ -0,0 +1,138 @@
+        ]
+
+    def run(self) -> pele_env.EnviroBuilder:
+        for arg in self.all_args:
+            arg_value = getattr(self.args, arg)
+            arg_value = atom_number_to_atom_string(self.original_system, arg_value)
+            new_atom_string = self.check_atom_string(arg_value)
+            setattr(self.args, arg, new_atom_string)
+        return self.args
+
+    def check_atom_string(self, args: List[str]) -> List[str]:
+        """
+        Checks if the atom string needs mapping by attempting to extract its coordinates.
+        """
+        output = []
+        args = args if isinstance(args, list) else [args]
+        for arg in args:
+            try:
+                helpers.get_coords_from_residue(self.original_system, arg)
+                helpers.get_coords_from_residue(self.preprocessed_pdb, arg)
+                output.append(arg)
+            except Exception as e:
+                self.logger.info("{} - mapping it now!".format(e))
+                _, after = self.map_atom_string(
+                    arg, self.original_system, self.preprocessed_pdb, self.logger
+                )
+                output.append(after)
+        return output
+
+    @staticmethod
+    def map_atom_string(
+        atom_string: str,
+        original_input: str,
+        preprocessed_file: str,
+        logger=Any,
+    ) -> (str, str):
+        """
+        Maps old atom string to a new atom string by comparing coordinates of the original and preprocessed PBD files.
+        """
+
+        # read in the original and preprocessed PDB lines
+        with open(original_input, "r") as initial:
+            initial_lines = initial.readlines()
+
+        with open(preprocessed_file, "r") as prep:
+            preprocessed_lines = prep.readlines()
+
+        # retrieve atom info from the original PDB
+        chain, resnum, atom_name = atom_string.split(":")
+
+        # extract coordinates from the original PDB
+        for i in initial_lines:
+            if (
+                (i.startswith("HETATM") or i.startswith("ATOM"))
+                and i[21].strip() == chain
+                and i[22:26].strip() == resnum
+                and i[12:16].strip() == atom_name
+            ):
+                initial_coords = get_coords_from_line(i)
+
+                # extract coordinates from preprocessed file and compare to the original one
+                for p in preprocessed_lines:
+                    preprocessed_coords = get_coords_from_line(p)
+
+                    if preprocessed_coords == initial_coords:
+                        new_atom_name, new_resnum, new_chain = get_atom_from_line(p)
+                        before = "{}:{}:{}".format(chain, resnum, atom_name)
+                        after = "{}:{}:{}".format(new_chain, new_resnum, new_atom_name)
+                        logger.info("Atom {} mapped to {}.".format(before, after))
+                        return before, after
+
+
+def get_atom_from_line(line):
+    """
```

Comment:
Apart from describing the function, it is always better to describe its parameters and returns. I would improve a little the docstrings in that sense.

Docstrings seem not that important but they are crucial if we want to offer a user-friendly API in the next versions.